### PR TITLE
ENH: also rename column name when only casing is different

### PR DESCRIPTION
**Pull request by KriWay-LV**
possible to rename column when only casing changes

### CODE REVIEW

File: geofileops/fileops.py [hunk 1]
```diff
@@ -760,6 +760,15 @@ def rename_column(
             raise ValueError(f"rename_column not supported for {path}")
 
         # Rename column
+        if column_name.lower() == new_column_name.lower():
+            sql_stmt = (
+                f'ALTER TABLE "{layer}" '
+                f'RENAME COLUMN "{column_name}" TO "{column_name}_"'
```
Comment (theroggy) [inline geofileops/fileops.py:766]: Probably safer to find a temporary column name in a loop while (case insensitively) checking that it doesn't exist yet in the list of current columns?

**Review by theroggy — CHANGES_REQUESTED** [review]
Can you also add an entry in the changelog...

File: geofileops/fileops.py [hunk 2]
```diff
@@ -760,6 +760,18 @@ def rename_column(
             raise ValueError(f"rename_column not supported for {path}")
 
         # Rename column
+        if column_name.lower() == new_column_name.lower():
+            temp_column_name = Path(tempfile.mktemp()).name
+            while temp_column_name in info.columns:
+                temp_column_name = Path(tempfile.mktemp()).name
```
Comment (theroggy) [inline geofileops/fileops.py:766]:
```suggestion
            for index in range(9999):
                temp_column_name = f"tmp_{index}"
                if temp_column_name not in info.columns:
                    break
```
Creating a temporary file for each try feels a bit weird... I think this is equivalent?

File: tests/test_geofile.py [hunk 3]
```diff
@@ -1071,11 +1071,17 @@ def test_rename_column(tmp_path, suffix):
     assert "OPPERVL" not in result_layerinfo.columns
     assert "area" in result_layerinfo.columns
 
+    # Rename column with different casing
```
Comment (theroggy) [inline tests/test_geofile.py:1074]: Best add a column to the test file so you trigger the code that avoids an equal temp column being created...

Comment (KriWay-LV) [reply tests/test_geofile.py:1074]: what do you mean? gfo.add_column first or just add a new column in the test file?

Comment (theroggy) [reply tests/test_geofile.py:1074]: Just use `gfo.add_column` to add it in the test...

Comment (theroggy) [reply tests/test_geofile.py:1074]: You can add the first tmp column in uppercase... then that has been tested as well... (e.g. TMP_0)

**Review by theroggy — APPROVED** [review]
I did some final, minor documentation changes, but this look good to go now.

Thanks!

**Pull request merged**